components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -16238,6 +16244,87 @@ public Set<String> getAPIVersions(String apiName, String apiProvider, String org
         return versions;
     }
 
+    public API getAPIInfoByNameVersion(String apiProvider, String apiName, String apiVersion, String organization) throws APIManagementException {
+        API api = new API(new APIIdentifier(apiProvider, apiName, apiVersion, ""));;
+        try (Connection connection = APIMgtDBUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(SQLConstants.GET_API_INFO_BY_NAME_VERSION)) {
+            statement.setString(1, APIUtil.replaceEmailDomainBack(apiProvider));
+            statement.setString(2, apiName);
+            statement.setString(3, organization);
+            statement.setString(4, apiVersion);
+            ResultSet resultSet = statement.executeQuery();
```

Comment:
Resultset needs to handle with try with resources

Reply:
Fixed with: https://github.com/wso2/carbon-apimgt/pull/12107/commits/86436c41fad866a8c62c80cdabe2a8953170e3be